src/figuresClasses.ts

```diff
@@ -2,10 +1,91 @@
+export interface Figure {
+  shape: `triangle` | `circle` | `rectangle`;
+  color: `red` | `green` | `blue`;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  shape: `triangle`;
+
+  constructor(
+    public color: `red` | `green` | `blue`,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = `triangle`;
+
+    const descSortedSides = [a, b, c].sort((first, second) => second - first);
+    const isCorrectSides =
+      descSortedSides[0] < descSortedSides[2] + descSortedSides[1];
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('The Length must be greater than 0');
+    }
+
+    if (!isCorrectSides) {
+      throw new Error(
+        'The Longest side of a triangle must be less than a sum of two others',
+      );
```

Comment:
[CODE STYLE] The error message could be more descriptive by specifying that it is the lengths of the triangle's sides that must satisfy the condition. For example, 'The longest side of a triangle must be less than the sum of the other two sides.'